snippets/ModalDialog.md

```diff
@@ -0,0 +1,135 @@
+### Modal/Dialog
+
+Renders a dialog component with modal.
+
+This is a simple React ModalDialog Component.It owns only one prop: `closeOnClick`.
+
+* `closeOnClick` prop determines whether we can close the dialog via clicking on modal.
+
+Benefits from the CustomEvent API,this ModalDialog component supports single instance usage.We just need import `Dialog` once time,and can call the dialog everywhere.Every time we wanna show the dialog,we just call the static function `Dialog.show`.Passing our jsx templates and data as the parameter and then the dialog will show with the data and templates rendered.
+
+After the component mounted,Dialog binds a custom event called 'modal' to document element.The event handler defined in our component named `modalHandler`.It accepts and renders the data passed by users.When document node captured a `modal` event,`modalHandler` works.
+
+Before the component unmounted,Dialog removes the custom event from document element in case of memory leaks.
+
+Dialog exposed the `show` method via using `static` reserved keyword so that we can call it through Dialog class.The `show` method accepts a object parameter that contains two properties:`title` and `content`.
```

Comment:
```suggestion
Dialog exposed the `show` method via using `static` reserved keyword so that we can call it through Dialog class.The `show` method accepts a object parameter that contains three properties: `title`, `closeOnClick`, and `content`.
```